Homework23 tests

ochaklosh opened the pull request:
This is a refactored Homework17Tests, addSongToPlaylist()
Several files were refactored to use POM with pageFactory
Methods were moved from the testpage to respective pages
Method getSuccessMsg() was refactored and renamed to be more universal in future use
Method waitForMsgToHide() was created to resolve a problem when the old value of the message is lacking time to be replaced by the new one.

ochaklosh committed:
Update ActionsTest.java

Still working

ochaklosh committed:
Update BasePage.java

Added method isPlaying()

ochaklosh committed:
Update HomePage.java

Still working

ochaklosh committed:
Update TestNG.xml

Updated which test to run

ochaklosh committed:
Update LoginTests.java

Currently with using PageFactoring Login does not work it gives mistake
Cannot invoke "org.openqa.selenium.WebElement.clear()" because "this.emailField" is null
which is a method to clear the emailField before entering users email

ochaklosh committed:
Create BasePage.java

Created BasePage to use with PageFactory

ochaklosh committed:
Created LoginPage

Created LoginPage to use pageFactory, so far unsuccessfully

ochaklosh committed:
Update AllSongsPage.java

Updated the find method from
findElementAndClickability   to
 waitForVisibility

ochaklosh committed:
Update HomePage.java

Updated by removing locator information within a method and add to the Locator section

ochaklosh committed:
Update TestNG.xml

Updated which test I want to run

ochaklosh committed:
Update BasePage.java

Updated waitForVisibilityPF and findElementAndClickability methods

to work with pageFactory t provide WebElement instead of a locator

ochaklosh committed:
Create HomePage.java

Added HomePage.java to pageFactory package
Updated locator userAvatarIcon to be used with @FindBy

ochaklosh committed:
Update LoginPage.java

Updated methods and imports to support PageFactory

ochaklosh committed:
Create Homework23Tests.java

Refactored addSongToPlaylist() from Homework17Tests using pageFactory

ochaklosh committed:
Create AllSongsPage.java

Created AllSongsPage using pageFactory within structured using POM

ochaklosh committed:
Update LoginTests.java

added space between the lines

ochaklosh committed:
Update HomePage.java

Added and refactored locators using @FindBy
Added methods createNewPlaylist(), searchSong(), waitForMsg().
Renamed getRenamePlaylistSuccessMsg() to
getSuccessMsg() - which can be used in other test to get the text of the Success message.

ochaklosh committed:
Update LoginPage.java

Added encapsulation to variables
Updated to use pageFactory

ochaklosh committed:
Update TestNG.xml

Added new test class

Berichicko commented:
Hi Oksana

You got this problem, please correct it.

![image](https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/cff5a737-b620-411a-b139-e2eeec6d2256)

![image](https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/0441a717-79c5-41c6-9aab-c666536ea1e9)

HINT

![image](https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/8c930308-30a4-4434-a63b-11e73b082d90)

```
 @FindBy(xpath = "//div[@class='success show' and contains(text(), 'Added 1 song into')]")
    private WebElement notificationSuccessMsg;
```

![image](https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/ad138c61-8856-46e8-a33a-4044de22130b)

ochaklosh committed:
Updated

Updated per instructors request a more detailed locator for notificationSuccessMsgOneSongAdded
using xpath and searching for text

Berichicko commented:
Hello,

I have checked your homework 23, and I wanted to let you know that you're doing great! Keep up the excellent work!
Best wishes from the HW team (Alex, [aliaksandr@testpro.io](mailto:aliaksandr@testpro.io))!

![Thumbs-up-hand-with-crown-illustration-on-transparent-background-PNG](https://github.com/Tootienahs/A59-QA-Automation-Framework/assets/104761482/20861dce-f8e7-4435-9b92-de6436933d0b)

![Quote](https://github-readme-quotes-bay.vercel.app/quote?theme=dark&layout=socrates)


![image](https://github.com/ochaklosh/A59-QA-Automation-Framework/assets/104761482/fbad5118-36b0-49be-b99a-55148dcf0323)
